src/main/java/jcifs/netbios/NbtAddress.java

```diff
@@ -0,0 +1,494 @@
+    byte[] macAddress;
+    String calledName;
+
+
+    NbtAddress ( Name hostName, int address, boolean groupName, int nodeType ) {
+        this.hostName = hostName;
+        this.address = address;
+        this.groupName = groupName;
+        this.nodeType = nodeType;
+    }
+
+
+    NbtAddress ( Name hostName, int address, boolean groupName, int nodeType, boolean isBeingDeleted, boolean isInConflict, boolean isActive,
+            boolean isPermanent, byte[] macAddress ) {
+
+        /*
+         * The NodeStatusResponse.readNodeNameArray method may also set this
+         * information. These two places where node status data is populated should
+         * be consistent. Be carefull!
+         */
+        this.hostName = hostName;
+        this.address = address;
+        this.groupName = groupName;
+        this.nodeType = nodeType;
+        this.isBeingDeleted = isBeingDeleted;
+        this.isInConflict = isInConflict;
+        this.isActive = isActive;
+        this.isPermanent = isPermanent;
+        this.macAddress = macAddress;
+        this.isDataFromNodeStatus = true;
+    }
+
+
+    /**
+     * {@inheritDoc}
+     *
+     * @see jcifs.Address#unwrap(java.lang.Class)
+     */
+    @SuppressWarnings ( "unchecked" )
+    @Override
+    public <T extends Address> T unwrap ( Class<T> type ) {
+        if ( type.isAssignableFrom(this.getClass()) ) {
+            return (T) this;
+        }
+        return null;
+    }
+
+
+    /**
+     * Guess next called name to try for session establishment. These
+     * methods are used by the smb package.
+     * 
+     * @return guessed name
+     */
+    @Override
+    public String firstCalledName () {
+
+        this.calledName = this.hostName.name;
+
+        if ( Character.isDigit(this.calledName.charAt(0)) ) {
+            int i, len, dots;
+            char[] data;
+
+            i = dots = 0; /* quick IP address validation */
+            len = this.calledName.length();
+            data = this.calledName.toCharArray();
+            while ( i < len && Character.isDigit(data[ i++ ]) ) {
+                if ( i == len && dots == 3 ) {
+                    // probably an IP address
+                    this.calledName = SMBSERVER_NAME;
+                    break;
+                }
+                if ( i < len && data[ i ] == '.' ) {
+                    dots++;
+                    i++;
+                }
+            }
+        }
+        else {
+            switch ( this.hostName.hexCode ) {
+            case 0x1B:
+            case 0x1C:
+            case 0x1D:
+                this.calledName = SMBSERVER_NAME;
+            }
+        }
+
+        return this.calledName;
+    }
+
+
+    /**
+     * 
+     * @param tc
+     *            context to use
+     * @return net name to try
+     */
+    @Override
+    public String nextCalledName ( CIFSContext tc ) {
+
+        if ( this.calledName == this.hostName.name ) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: An equality operator (== or !=) is used to compare strings, which matches strings based on address.

**Fix**: Do content comparison using the equals() method to compare the values of the strings.